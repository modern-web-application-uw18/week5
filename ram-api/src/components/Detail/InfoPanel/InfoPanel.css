.InfoPanel {
    height: 90%;
    width: 300px;
    top: 5%;
    background-color: rgba(0, 0, 0, 0.5);
    position: relative;
    margin: auto;
    color: #ffffff;
    border-radius: 5%;
    
}

.InfoPanelDetail {
    font-size: 15px;
    color: #6e6e6e94;
    display: block;
    margin: 10px;
}

.Star {
    font-size: 40px;
    text-align: right;
    top: 0px;
    right: 5px;
    position: absolute;
    cursor: pointer;
}


.Star:hover,
.Star:active,
.Star.active {
    color: #e09900;

    
}