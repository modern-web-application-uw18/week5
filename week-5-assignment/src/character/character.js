import React, {Component} from 'react';
import '../App.css'
import 'bootstrap/dist/css/bootstrap.css';
import PropTypes from 'prop-types';

class Character extends Component{
    constructor(props){
        super(props);
        this.state = {
            character: this.character
        };
    }

    character = this.props.match.params.number;

    componentDidMount(){
        fetch(`https://swapi.co/api/people/?search=${this.character}`)
        .then(response => response.json())
        .then(json => this.setState((prevState) => {
            return {character: json.results};
    }));
    }

    

    render(){
    return (
      <div className="border border-success bg-light">
        <p>Name: {this.state.character[0].name}</p>
        <p>Height: {this.state.character[0].height}</p>
        <p>Mass: {this.state.character[0].mass}</p>
        <p>Birth Year: {this.state.character[0].birth_year}</p>
        <p>Gender: {this.state.character[0].gender}</p>
      </div>
    )
  }
}

Character.propTypes = {
        name: PropTypes.string,
        height: PropTypes.string,
        mass: PropTypes.string,
        birth_year: PropTypes.string,
        gender: PropTypes.string
}

export default Character;