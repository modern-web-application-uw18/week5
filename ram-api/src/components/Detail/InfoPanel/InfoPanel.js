import React, { Component } from 'react';
import './InfoPanel.css';
import PropTypes from 'prop-types';
import MdAccessibility from 'react-icons/lib/md/accessibility';
import MdAssignmentInd from 'react-icons/lib/md/assignment-ind';
import MdBugReport from 'react-icons/lib/md/bug-report';
import MdLocationOn from 'react-icons/lib/md/location-on';
import MdStarBorder from 'react-icons/lib/md/star-border';
import MdStar from 'react-icons/lib/md/star';

class InfoPanel extends Component {

    state = {
        clicked: JSON.parse(localStorage.getItem("favoriteIds")).indexOf(this.props.character.id) !== -1 ? true : false
    }





    starHandler = () => {
        this.props.favAddRemoveHandler(this.props.character.id);
        let clicked = !this.state.clicked;
        this.setState(prevState => {
            return {
                clicked: clicked
            }
        })

    }

    componentDidUpdate(prevProps, prevState) {
        let storageClicked = JSON.parse(localStorage.getItem("favoriteIds")).indexOf(this.props.character.id) !== -1 ? true : false;
        if (prevState.clicked !== storageClicked) {
            this.setState((prevState) => {
                return {
                    clicked: storageClicked
                }
            })
        }
    }


    render() {



        return (
            <div className="InfoPanel" >
                <img src={this.props.character.image} style={{ borderRadius: '5%' }} alt={this.props.character.name} />

                {this.state.clicked ?
                    <div className="Star" onClick={this.starHandler}><MdStar style={{ color: '#e09900' }} /> </div>
                    :
                    <div className="Star" onClick={this.starHandler}><MdStarBorder /> </div>}

                <br />
                <br />
                <h3><span className="label label-danger">{this.props.character.name}</span></h3>
                <br />
                <span className="InfoPanelDetail"><MdAccessibility size={40} /> {this.props.character.gender ? this.props.character.gender : ''}</span>
                <span className="InfoPanelDetail"><MdAssignmentInd size={40} /> {this.props.character.status ? this.props.character.status : ''}</span>
                <span className="InfoPanelDetail"><MdBugReport size={40} /> {this.props.character.species ? this.props.character.species : ''}</span>
                <span className="InfoPanelDetail"><MdLocationOn size={40} />{this.props.character.location ? this.props.character.location.name : ''}</span>
            </div>
        )
    }

}

InfoPanel.propTypes = {
    character: PropTypes.object.isRequired,
    favAddRemoveHandler: PropTypes.func
}


export default InfoPanel;