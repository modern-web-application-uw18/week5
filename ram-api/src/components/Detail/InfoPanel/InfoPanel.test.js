import React from 'react';
import { mount } from 'enzyme';
import InfoPanel from './InfoPanel';
import character from '../../../__mocks__/character.json'

describe('<InfoPanel />', () => {
    it('Should render InfoPanel', () => {
        const wrapper = mount(<InfoPanel character={character} />);      
        expect(wrapper).toMatchSnapshot();
    })
})