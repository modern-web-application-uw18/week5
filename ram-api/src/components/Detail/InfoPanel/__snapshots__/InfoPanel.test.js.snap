// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoPanel /> Should render InfoPanel 1`] = `
<InfoPanel
  character={
    Object {
      "created": "2017-11-04T19:09:56.428Z",
      "episode": Array [
        "https://rickandmortyapi.com/api/episode/6",
        "https://rickandmortyapi.com/api/episode/7",
        "https://rickandmortyapi.com/api/episode/8",
        "https://rickandmortyapi.com/api/episode/9",
        "https://rickandmortyapi.com/api/episode/10",
        "https://rickandmortyapi.com/api/episode/11",
        "https://rickandmortyapi.com/api/episode/12",
        "https://rickandmortyapi.com/api/episode/14",
        "https://rickandmortyapi.com/api/episode/15",
        "https://rickandmortyapi.com/api/episode/16",
        "https://rickandmortyapi.com/api/episode/17",
        "https://rickandmortyapi.com/api/episode/18",
        "https://rickandmortyapi.com/api/episode/19",
        "https://rickandmortyapi.com/api/episode/20",
        "https://rickandmortyapi.com/api/episode/21",
        "https://rickandmortyapi.com/api/episode/22",
        "https://rickandmortyapi.com/api/episode/23",
        "https://rickandmortyapi.com/api/episode/24",
        "https://rickandmortyapi.com/api/episode/25",
        "https://rickandmortyapi.com/api/episode/26",
        "https://rickandmortyapi.com/api/episode/27",
        "https://rickandmortyapi.com/api/episode/29",
        "https://rickandmortyapi.com/api/episode/30",
        "https://rickandmortyapi.com/api/episode/31",
      ],
      "gender": "Female",
      "id": 3,
      "image": "https://rickandmortyapi.com/api/character/avatar/3.jpeg",
      "location": Object {
        "name": "Earth (Replacement Dimension)",
        "url": "https://rickandmortyapi.com/api/location/20",
      },
      "name": "Summer Smith",
      "origin": Object {
        "name": "Earth (Replacement Dimension)",
        "url": "https://rickandmortyapi.com/api/location/20",
      },
      "species": "Human",
      "status": "Alive",
      "type": "",
      "url": "https://rickandmortyapi.com/api/character/3",
    }
  }
>
  <div
    className="InfoPanel"
  >
    <img
      alt="Summer Smith"
      src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
      style={
        Object {
          "borderRadius": "5%",
        }
      }
    />
    <div
      className="Star"
      onClick={[Function]}
    >
      <MdStarBorder>
        <IconBase
          viewBox="0 0 40 40"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m20 25.7l6.3 3.8-1.7-7.2 5.6-4.8-7.4-0.6-2.8-6.7-2.8 6.7-7.4 0.6 5.6 4.8-1.6 7.2z m16.6-10.3l-9 7.9 2.7 11.7-10.3-6.2-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11z"
              />
            </g>
          </svg>
        </IconBase>
      </MdStarBorder>
       
    </div>
    <br />
    <br />
    <h3>
      <span
        className="label label-danger"
      >
        Summer Smith
      </span>
    </h3>
    <br />
    <span
      className="InfoPanelDetail"
    >
      <MdAccessibility
        size={40}
      >
        <IconBase
          size={40}
          viewBox="0 0 40 40"
        >
          <svg
            fill="currentColor"
            height={40}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width={40}
          >
            <g>
              <path
                d="m35 15h-10v21.6h-3.4v-10h-3.2v10h-3.4v-21.6h-10v-3.4h30v3.4z m-15-11.6c1.8 0 3.4 1.4 3.4 3.2s-1.6 3.4-3.4 3.4-3.4-1.6-3.4-3.4 1.6-3.2 3.4-3.2z"
              />
            </g>
          </svg>
        </IconBase>
      </MdAccessibility>
       
      Female
    </span>
    <span
      className="InfoPanelDetail"
    >
      <MdAssignmentInd
        size={40}
      >
        <IconBase
          size={40}
          viewBox="0 0 40 40"
        >
          <svg
            fill="currentColor"
            height={40}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width={40}
          >
            <g>
              <path
                d="m30 31.6v-2.3c0-3.4-6.6-5.2-10-5.2s-10 1.8-10 5.2v2.3h20z m-10-20c-2.7 0-5 2.3-5 5s2.3 5 5 5 5-2.2 5-5-2.3-5-5-5z m0-6.6c-0.9 0-1.6 0.7-1.6 1.6s0.7 1.8 1.6 1.8 1.6-0.8 1.6-1.8-0.7-1.6-1.6-1.6z m11.6 0c1.8 0 3.4 1.6 3.4 3.4v23.2c0 1.8-1.6 3.4-3.4 3.4h-23.2c-1.8 0-3.4-1.6-3.4-3.4v-23.2c0-1.8 1.6-3.4 3.4-3.4h6.9c0.7-2 2.5-3.4 4.7-3.4s4 1.4 4.7 3.4h6.9z"
              />
            </g>
          </svg>
        </IconBase>
      </MdAssignmentInd>
       
      Alive
    </span>
    <span
      className="InfoPanelDetail"
    >
      <MdBugReport
        size={40}
      >
        <IconBase
          size={40}
          viewBox="0 0 40 40"
        >
          <svg
            fill="currentColor"
            height={40}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width={40}
          >
            <g>
              <path
                d="m23.4 20v-3.4h-6.8v3.4h6.8z m0 6.6v-3.2h-6.8v3.2h6.8z m10-13.2v3.2h-3.6c0.1 0.6 0.2 1.2 0.2 1.8v1.6h3.4v3.4h-3.4v1.6c0 0.5-0.1 1.1-0.2 1.6h3.6v3.4h-4.7c-1.7 3-5 5-8.7 5s-7-2-8.7-5h-4.7v-3.4h3.6c-0.1-0.5-0.2-1.1-0.2-1.6v-1.6h-3.4v-3.4h3.4v-1.6c0-0.6 0.1-1.2 0.2-1.8h-3.6v-3.2h4.7c0.8-1.4 1.8-2.5 3.1-3.3l-2.8-2.8 2.4-2.3 3.7 3.6c0.7-0.2 1.5-0.2 2.3-0.2s1.6 0 2.3 0.2l3.7-3.6 2.4 2.3-2.8 2.8c1.3 0.8 2.3 1.9 3.1 3.3h4.7z"
              />
            </g>
          </svg>
        </IconBase>
      </MdBugReport>
       
      Human
    </span>
    <span
      className="InfoPanelDetail"
    >
      <MdLocationOn
        size={40}
      >
        <IconBase
          size={40}
          viewBox="0 0 40 40"
        >
          <svg
            fill="currentColor"
            height={40}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width={40}
          >
            <g>
              <path
                d="m20 19.1q1.7 0 2.9-1.2t1.2-2.9-1.2-2.9-2.9-1.2-2.9 1.2-1.2 2.9 1.2 2.9 2.9 1.2z m0-15.7q4.8 0 8.2 3.4t3.4 8.2q0 2.4-1.2 5.5t-2.9 5.9-3.4 5.1-2.8 3.8l-1.3 1.3q-0.5-0.5-1.2-1.4t-2.9-3.6-3.5-5.2-2.8-5.8-1.2-5.6q0-4.8 3.4-8.2t8.2-3.4z"
              />
            </g>
          </svg>
        </IconBase>
      </MdLocationOn>
      Earth (Replacement Dimension)
    </span>
  </div>
</InfoPanel>
`;
